packages/compiler-cli/src/ngtsc/partial_evaluator/src/ts_helpers.ts

```diff
@@ -35,3 +35,27 @@ export class SpreadHelperFn extends KnownFn {
     return result;
   }
 }
+
+// Used for `__spreadArray` TypeScript helper function.
+export class SpreadArrayHelperFn extends KnownFn {
+  evaluate(node: ts.Node, args: ResolvedValueArray): ResolvedValue {
+    if (args.length !== 2) {
+      return DynamicValue.fromUnknown(node);
+    }
+
+    const [to, from] = args;
+    if (to instanceof DynamicValue) {
+      return DynamicValue.fromDynamicInput(node, to);
+    } else if (from instanceof DynamicValue) {
+      return DynamicValue.fromDynamicInput(node, from);
+    }
+
+    if (!Array.isArray(to)) {
+      return DynamicValue.fromInvalidExpressionType(node, to);
+    } else if (!Array.isArray(from)) {
+      return DynamicValue.fromInvalidExpressionType(node, from);
+    }
+
+    return from.concat(to);
```

Comment:
Oops, I spoke too soon. This should be

```suggestion
    return to.concat(from);
```

Reply:
That is entirely correct! I didn't notice as the evaluator spec are part of `ngtsc/partial_evaluator` and I only ran the `ngcc` test targets.